src/golomb.h

```diff
@@ -33,11 +33,16 @@
 namespace pg::golomb
 {
 
+template<typename InputIt>
+concept integral_iterator = std::integral< typename std::iterator_traits< InputIt >::value_type >;
+
+template<typename InputIt>
+concept unsigned_integral_iterator = std::unsigned_integral< typename std::iterator_traits< InputIt >::value_type >;
+
```

Comment:
These these concepts should be moved into the detail namespace since these are not part of the API.

Reply:
Done, in MSVC at first it required the std:: to be specified as global namespace (::std::), but it turned out to be a messup.
Code now compiles in MSVC and G++ (required extra change in the file, see other comment)